pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java

```diff
@@ -95,6 +95,8 @@
         public static final String REALTIME_SEGMENT_FLUSH_TIME = "realtime.segment.flush.threshold.time";
         // Num records threshold in the realtime segment
         public static final String REALTIME_SEGMENT_FLUSH_SIZE = "realtime.segment.flush.threshold.size";
+        // Flush threshold size across all segments
```

Comment:
It's also not a hard threshold, which makes it even more complicated to explain.

For example, imagine you that there is one Kafka partition served by this server and that the threshold is set to one million. If the Kafka partition count is increased to 10 while this segment is still open, this means that there will be up to 1.9 million rows in memory (1M + 9 * 100k rows for the new partitions).